### PR TITLE
More functions implemented, few bugfixes

### PR DESCRIPTION
**Pull request by doiiido**
Another crew worked on this for a semester....

### CODE REVIEW

File: samples/java/MinMax1.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import java.util.Scanner;
+
+/*
+ * "Ler 3 números inteiros da entrada, imprimir o menor e o maior"
+ *
+ * Autor: Jacques Sauvé
```
Comment (claudiosegala) [inline samples/java/MinMax1.java]:
```suggestion

```

File: src/include/class_loader/field.hpp [hunk 2]
```diff
@@ -16,6 +19,21 @@ namespace jvm {
 		 */
 		uint16_t name_index;
 
+		//TODO THIS IS NOT RIGHT, BUT IT WORKS FOR SIMPLE CLASSES
```
Comment (claudiosegala) [inline src/include/class_loader/field.hpp:22]: @doiiido fair enough

File: src/lib/class_loader/class_loader.cpp [hunk 3]
```diff
@@ -1,223 +1,270 @@
...
+
+	void ClassLoader::read_cp (Reader& file) {
+		cp_count = (uint16_t)(file.getNextHalfWord() - 1);
+
+		if (cp_count <= 0) {
+			return;
+		}
+
+		constant_pool.fill(file, cp_count);
+		cp_count ++;
```
Comment (claudiosegala) [inline src/lib/class_loader/class_loader.cpp:72]: @doiiido is this correct? Was this wrong in our JVM?

File: src/lib/class_loader/instruction_info.cpp [hunk 4]
```diff
@@ -4187,9 +4194,8 @@ namespace jvm {
 		if(!count)
 			throw JvmException("Invalid invokeinterface: the value of count must not be zero");
 
-		if(!data[idx+4])
+		if(data[idx+4] != 0)
```
Comment (claudiosegala) [inline src/lib/class_loader/instruction_info.cpp:4197]: @doiiido isn't this the same thing?

File: src/lib/class_loader/instruction_info.cpp [hunk 5]
```diff
@@ -4213,10 +4219,10 @@ namespace jvm {
 
 	uint32_t OPINFOinvokedynamic::fillParams (uint32_t idx, std::vector<u1>& data) {
 		index = Converter::to_u2(data[idx+1], data[idx+2]);
-		if(!data[idx+3] || !data[idx+4]) {
+		if(data[idx+3] != 0 || data[idx+4] != 0) {
```
Comment (claudiosegala) [inline src/lib/class_loader/instruction_info.cpp:4222]: @doiiido isn't this the same thing?

File: src/lib/util/reader.cpp [hunk 6]
```diff
@@ -1,90 +1,81 @@
...
-
-		// reserve in memory the space needed for the file
-		bytes.resize(size());
-
-		// read the file to memory
-		for (auto i = 0; !_class.eof(); i++) {
-			_class.read((char*) &bytes[i], 1);
-		}
-
-		if (!isValid()) {
```
Comment (claudiosegala) [inline src/lib/util/reader.cpp:35]: @doiiido why remove this from here?

**Comment by ramkury** [comment]
Is the "build" folder useful at all? Most of its files just look like clutter, they should be removed.

**Comment by doiiido** [comment]
@ramkury they can be removed, they are just the results of script execution

**Comment by ramkury** [comment]
> @ramkury they can be removed, they are just the results of script execution

In that case, it's a good idea to add the folder to .gitignore

**Review by claudiosegala — COMMENTED** [review]
@ramkury this sounds fine to me. What do you think?

File: src/lib/class_loader/instruction_info.cpp (hunk 4 above)
Comment (ramkury) [reply src/lib/class_loader/instruction_info.cpp:4197]: @claudiosegala Exactly the opposite. The new behaviour seems to be the correct one, according to https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.invokeinterface

File: src/lib/class_loader/instruction_info.cpp (hunk 5 above)
Comment (ramkury) [reply src/lib/class_loader/instruction_info.cpp:4222]: @claudiosegala Exactly the opposite. The new behaviour seems to be the correct one, according to https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.invokedynamic

File: src/lib/class_loader/instruction_info.cpp (hunk 4 above)
Comment (claudiosegala) [reply src/lib/class_loader/instruction_info.cpp:4197]: That was dumb kkkkk

File: src/lib/class_loader/instruction_info.cpp (hunk 5 above)
Comment (claudiosegala) [reply src/lib/class_loader/instruction_info.cpp:4222]: That was dumb

File: src/lib/util/reader.cpp (hunk 6 above)
Comment (doiiido) [reply src/lib/util/reader.cpp:35]: Moved to class_loader.cpp line 96 to mimic the behavior of the oracle's own jvm and not load the class file if the magic number does not match.

File: src/lib/class_loader/class_loader.cpp (hunk 3 above)
Comment (doiiido) [reply src/lib/class_loader/class_loader.cpp:72]: There were a few changes, but git recognized the entire file as changed, I don't know how to fix it yet.
Line 72 was a quick fix at commit [4b3ea8a104232a43c5bbb1b95bd10f90c62b57a7](https://github.com/doiiido/jvm/commit/4b3ea8a104232a43c5bbb1b95bd10f90c62b57a7)

**Pull request merged**